ragaai_catalyst/redteaming/red_teaming.py

```
@@ -0,0 +1,329 @@
+        print(f"\nResults saved to: {save_path}")
+        return save_path
+
+    def _run_with_examples(self, description: str, detectors: List[str], response_model: Any, examples: List[str], scenarios_per_detector: int) -> pd.DataFrame:
+        results = []
+        # Process each detector
+        for detector in detectors:
+            print('='*50)
+            print(f"Running detector: [yellow2]{detector}[/yellow2]")
+            print('='*50)
+
+            if type(detector) == str:
```

Ruff (0.8.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)